packages/browser-destinations/src/destinations/intercom/trackEvent/index.ts

```diff
@@ -0,0 +1,91 @@
+import type { BrowserActionDefinition } from '../../../lib/browser-destinations'
+import { Intercom } from '../api'
+import type { Settings } from '../generated-types'
+import { filterCustomTraits, isEmpty } from '../utils'
+import type { Payload } from './generated-types'
+
+const action: BrowserActionDefinition<Settings, Intercom, Payload> = {
+  title: 'Track Event',
+  description: 'Submit an event to Intercom.',
+  platform: 'web',
+  fields: {
+    event_name: {
+      description: 'The name of the event.',
+      label: 'Event Name',
+      type: 'string',
+      required: true,
+      default: {
+        '@path': '$.event'
+      }
+    },
+    revenue: {
+      description:
+        'The amount associated with a purchase. Segment will multiply by 100 as Intercom requires the amount in cents.',
+      label: 'Revenue',
+      type: 'number',
+      required: false,
+      default: {
+        '@path': '$.properties.revenue'
+      }
+    },
+    currency: {
+      description:
+        'The currency of the purchase amount. Segment will default to USD if revenue is provided without a currency.',
+      label: 'Currency',
+      type: 'string',
+      required: false,
+      default: {
+        '@path': '$.properties.currency'
+      }
+    },
+    event_metadata: {
+      description: 'Optional metadata describing the event.',
+      label: 'Event Metadata',
+      type: 'object',
+      required: true,
+      default: {
+        '@path': '$.properties'
+      }
+    }
+  },
+  perform: (Intercom, event) => {
+    //remove event_name & event_metadata from the payload (they will be handled separately)
+    const { event_name, event_metadata, ...rest } = event.payload
+    const payload = { ...rest }
+    const richLinkProperties = Intercom.richLinkProperties ? Intercom.richLinkProperties : []
+
+    // create a list of the richLinkObjects that will be passed to Intercom
+    const richLinkObjects: { [k: string]: unknown } = {}
+    if (richLinkProperties.length != 0) {
+      Object.entries(event_metadata).forEach(([key, value]) => {
+        if (richLinkProperties.includes(key)) {
+          richLinkObjects[key] = value
+        }
+      })
+    }
+
+    // filter out reserved fields, drop custom objects & arrays
+    const filteredMetadata = filterCustomTraits(event_metadata)
+
+    // create price object
+    let price = {}
+    if (payload.revenue) {
+      price = {
+        amount: payload.revenue * 100,
+        currency: payload.currency ?? 'USD'
+      }
```

Comment:
Not a blocker, but the below may be very minutely more efficient because you don't have to create a new object and overwrite the existing one initialized on line 71.

```suggestion
      price.amount = payload.revenue * 100
      price.currency = payload.currency ?? 'USD'
```

Reply:
the only problem with this is that `amount` and `currency` dont exist on the type {}. 

I could define an interface `Price` that has keys `amount?` & `currency?`? And define `price` like:

const price : Price = {}

@brennan

Reply:
Makes sense - I would just ignore the comment, then, and let's move forward! 🚢